### PR TITLE
Add Authentication and Authorization to the app

### PR DESCRIPTION
**Pull request by ChubaOraka**
This pull request enables the users to register for an account in order to access the data visualizations on the main site.

### CODE REVIEW

File: app/db.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import streamlit as st
+from sqlalchemy.sql import text
+
+conn = st.experimental_connection('REPLACE-WITH-DATABASE-NAME', type='sql')
+
+# Retrieve all user's data from database
+def get_all_users():
+    with conn.session as s:
+        s.execute(text('CREATE TABLE IF NOT EXISTS drea_users (username varchar(255) PRIMARY KEY, email TEXT, name TEXT, passwordhash TEXT, approved BOOLEAN);'))
```
Comment (mattzamora) [inline app/db.py:9]: Use "not null" to declare which columns are required.

File: app/db.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import streamlit as st
+from sqlalchemy.sql import text
+
+conn = st.experimental_connection('REPLACE-WITH-DATABASE-NAME', type='sql')
```
Comment (mattzamora) [inline app/db.py:4]: Can this be replaced with a config variable?

File: app/views/login.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        except Exception as e:
+            st.error(e)
+    else:
+        name, authentication_status, username = authenticator.login('Login', 'main')
+        if not authentication_status:
+            st.button(':blue[Sign Up instead]', on_click=toggle_signup_login)
+
+    if not st.session_state.signup and "authentication_status" in st.session_state:
+        if st.session_state["authentication_status"]:
+            authenticator.logout('Logout', 'main')
```
Comment (mattzamora) [inline app/views/login.py:50]: Move "Log out" button to the sidebar over the heading "Compounds"

File: app/db.py (hunk 1 above)
Comment (ChubaOraka) [reply app/db.py]: [Done](https://github.com/MoCoMakers/nf_streamlit/commit/ba2bd8ef0a9b6b710ec12764195518fec1aa82d0#diff-2b5515043a2bd4e12be4d1847276b542fcfeaa6232c7d7a655dedab138b32e48R9)

File: app/db.py (hunk 2 above)
Comment (ChubaOraka) [reply app/db.py]: Yeah, the placeholder was a little misleading. [The replacement code](https://github.com/MoCoMakers/nf_streamlit/commit/e2b3baa77ef1f01df0d6cbd6bcd0d6e85178148d#diff-2b5515043a2bd4e12be4d1847276b542fcfeaa6232c7d7a655dedab138b32e48R4) is a more accurate.

https://github.com/MoCoMakers/nf_streamlit/blob/e2b3baa77ef1f01df0d6cbd6bcd0d6e85178148d/app/db.py#L4

File: app/views/login.py (hunk 3 above)
Comment (ChubaOraka) [reply app/views/login.py]: The "Logout" button has been [moved to the sidebar](https://github.com/MoCoMakers/nf_streamlit/commit/e26c9e3b19468cbb3bd801f68fcc22eea7a907cd#diff-e85e48b78e62649b467147e0fbeaef79cdd133e2442dfc22d0fde62cee31fb02R50).

https://github.com/MoCoMakers/nf_streamlit/blob/e26c9e3b19468cbb3bd801f68fcc22eea7a907cd/app/views/login.py#L50

File: app/db.py (hunk 1 above)
Comment (mattzamora) [reply app/db.py]: Looks good

**Pull request merged**